plugins/radare2/radare2_main.ml

```diff
@@ -0,0 +1,88 @@
+open Bap_core_theory
+open Core_kernel
+open Bap_future.Std
+open Bap.Std
+include Self()
+
+open KB.Syntax
+
+let agent =
+  KB.Agent.register ~package:"bap.std" "radare2-symbolizer"
+    ~desc:"extracts symbols radare2"
+
+let provide_roots funcs =
+  let promise_property slot =
+    KB.promise slot @@ fun label ->
+    KB.collect Theory.Label.addr label >>| function
+    | None -> None
+    | Some addr ->
+      let addr = Bitvec.to_bigint addr in
+      Option.some_if (Hashtbl.mem funcs addr) true in
+  promise_property Theory.Label.is_valid;
+  promise_property Theory.Label.is_subroutine
+
+let extract_name (json : Yojson.t) = 
+  match json with
+  | Yojson.(`Assoc list) -> 
+    (match List.find list ~f:(fun (key, _) -> String.equal key "name") with
+     | Some (_, v) -> (match v with
+         | Yojson.(`String str) -> Some str
+         | _ -> None)
+     | _ -> None)
+  | _ -> None
+
+let extract_addr (json : Yojson.t) = 
+  match json with
+  | Yojson.(`Assoc list) -> 
```

Comment:
also, there is no need to use the local open construct, `Module.(expr)`, here even if the constructor would be a regular one, e.g., you can write `Yojson.Assoc list` and it will work (if we will imagine that Yojson uses regular variants instead of polymorphic). But writing `Yojson.(Assoc list)` is a little bit different, if `Yojson` will define a function named `list` this expression will be `Yojson.Assoc Yojson.list`, which will be very different from what you want.